spccore/internal/lock.py

```diff
@@ -5,6 +5,26 @@
 
 from spccore.internal.dozer import *
 
+"""
+An implementation of a lock system that use the os file system as a single semaphore.
+
+In this implementation, to acquire a lock, we attempt to create a file in the os file system. The os will ensure that
+only one file is created at a time. To release a lock, we attempt to remove the file.
+
+Example::
+    user1_lock = Lock("foo", max_age=datetime.timedelta(seconds=5))
+    user2_lock = Lock("foo", max_age=datetime.timedelta(seconds=5))
+
+    with user1_lock:
```

Comment:
indicate threads are involved. sequentially executed, the second one will not wait